### PR TITLE
Add info about GSoC 2015, 2016

### PR DESCRIPTION
**Pull request by gxyd**
For info about students I saw the following posts:
- for gsoc 2016 students https://groups.google.com/forum/#!topic/sympy/szrD_obTYCs
- for gsoc 2015 students https://groups.google.com/forum/#!topic/sympy/LOubPFEr5jw

### CODE REVIEW

File: doc/src/aboutus.rst [hunk 1]
```diff
@@ -24,6 +24,7 @@ Financial and Infrastructure Support
   * in 2012 for 6 students (`GSoC 2012 <https://github.com/sympy/sympy/wiki/Gsoc-2012-report>`_)
   * in 2013 for 7 students (`GSoC 2013 <https://github.com/sympy/sympy/wiki/Gsoc-2013-report>`_)
   * in 2014 for 10 students (`GSoC 2014 <https://github.com/sympy/sympy/wiki/Gsoc-2014-report>`_)
+  * in 2016 for 8 students (`GSoC 2016 <https://github.com/sympy/sympy/wiki/GSoC-2016-report>`_)
```
Comment (asmeurer) [inline doc/src/aboutus.rst]: If you are counting James and Rajith that's 9 students. SymPy as a mentoring org had 7 students (originally 8 but one dropped out).

Comment (gxyd) [inline doc/src/aboutus.rst]: Okay, I thought so, since I never read about that project. I will change it to "7 students".

**Comment by gxyd** [comment]
I have force pushed and addressed the comment.

**Comment by asmeurer** [comment]
There's some double counting here for the years that SymPy wasn't a mentoring org, which is a bit confusing. Also, the lower bullet points should be indented, as they are part of GSoC.

File: doc/src/aboutus.rst [hunk 2]
```diff
@@ -24,6 +24,7 @@ Financial and Infrastructure Support
   * in 2012 for 6 students (`GSoC 2012 <https://github.com/sympy/sympy/wiki/Gsoc-2012-report>`_)
   * in 2013 for 7 students (`GSoC 2013 <https://github.com/sympy/sympy/wiki/Gsoc-2013-report>`_)
   * in 2014 for 10 students (`GSoC 2014 <https://github.com/sympy/sympy/wiki/Gsoc-2014-report>`_)
+  * in 2016 for 7 students (`GSoC 2016 <https://github.com/sympy/sympy/wiki/GSoC-2016-report>`_)
```
Comment (asmeurer) [inline doc/src/aboutus.rst]: Maybe this top list should have complete totals (e.g., 9 for this year), and then below it, we can write "of these, we would like to thank these organizations for hosting GSoC students under their umbrella organizations:"

Comment (asmeurer) [inline doc/src/aboutus.rst]: And then keep everything below, except indented one level

File: doc/src/aboutus.rst [hunk 3]
```diff
@@ -42,6 +45,8 @@ Financial and Infrastructure Support
 
 * `The Space Telescope Science Institute <http://www.stsci.edu/portal/>`_: STScI hosted 1 GSoC 2007 student (Mateusz)
 
+* `The Ruby Science Foundation <http://www.sciruby.com/>`_: SciRuby hosted 1 GSoC 2016 student (Rajith)
```
Comment (asmeurer) [inline doc/src/aboutus.rst]: That is, everything up to this. The stuff below this is unrelated to GSoC.

**Comment by gxyd** [comment]
I am now adding "SymPy's turn" label, just updated it.

File: doc/src/aboutus.rst [hunk 4]
```diff
@@ -24,23 +24,32 @@ Financial and Infrastructure Support
   * in 2012 for 6 students (`GSoC 2012 <https://github.com/sympy/sympy/wiki/Gsoc-2012-report>`_)
   * in 2013 for 7 students (`GSoC 2013 <https://github.com/sympy/sympy/wiki/Gsoc-2013-report>`_)
   * in 2014 for 10 students (`GSoC 2014 <https://github.com/sympy/sympy/wiki/Gsoc-2014-report>`_)
+  * in 2015 for 7 students (`GSoC 2014 <https://github.com/sympy/sympy/wiki/Gsoc-2015-report>`_)
```
Comment (asmeurer) [inline doc/src/aboutus.rst]: The docs build failed. This should be "2015", not "2014".

**Comment by asmeurer** [comment]
This looks good once you fix the build error.

**Comment by gxyd** [comment]
Changed 2014 -> 2015.

File: doc/src/aboutus.rst [hunk 5]
```diff
@@ -24,23 +24,32 @@ Financial and Infrastructure Support
...
+    * 3 students (Aaron, Dale and Fabian) in 2009
+    * 1 student (Addison) in 2010
 
-  * 1 student (Chris) in 2007
-  * 3 students (Aaron, Dale and Fabian) in 2009
-  * 1 student (Addison) in 2010
+  * `The Space Telescope Science Institute <http://www.stsci.edu/portal/>`_: STScI hosted 1 GSoC 2007 student (Mateusz)
 
-* `The Space Telescope Science Institute <http://www.stsci.edu/portal/>`_: STScI hosted 1 GSoC 2007 student (Mateusz)
+  * `The Ruby Science Foundation <http://www.sciruby.com/>`_: SciRuby hosted 1 GSoC 2016 student (Rajith)
```
Comment (isuruf) [inline doc/src/aboutus.rst]: There was @abinashmeher999 in 2015

**Comment by gxyd** [comment]
Is there anyone else I may have missed, apart from the one mentioned by @isuruf

File: doc/src/aboutus.rst (hunk 5 above)
Comment (abinashmeher999) [inline doc/src/aboutus.rst]: Thanks for pointing it out :simple_smile:

File: doc/src/aboutus.rst [hunk 6]
```diff
@@ -24,23 +24,36 @@ Financial and Infrastructure Support
...
+  umbrella organizations:
+
+  * `Python Software Foundation (PSF) <https://www.python.org/psf/>`_ has hosted
+    various GSoC students over the years:
+
+    * 3 GSoC 2007 students (Brian, Robert and Jason)
+    * 1 GSoC 2008 student (Fredrik)
+    * 2 GSoC 2009 students (Freddie and Priit)
+    * 4 GSoC 2010 students (Aaron, Christian, Matthew and Øyvind)
+    * 7 GSoC 2015 students
```
Comment (isuruf) [inline doc/src/aboutus.rst]: This would become 6.

Comment (gxyd) [inline doc/src/aboutus.rst]: Looking at the GSoC 2015 report https://github.com/sympy/sympy/wiki/Gsoc-2015-report it only says 5 students. Should it be 5 then? Or is that report missing something?
BTW this number here represents the total number of GSoC students (participated under SymPy + some umbrella organisation).

Comment (gxyd) [inline doc/src/aboutus.rst]: I can see that @leosartaj is missing from that GSoC 2015 report.

Comment (gxyd) [inline doc/src/aboutus.rst]: IIRC then there was project by @debugger22 on assumptions. That would make it 7. So I think the current number `7` is correct.

Comment (asmeurer) [inline doc/src/aboutus.rst]: There were only 6 PSF GSoC projects in 2015 (https://www.google-melange.com/archive/gsoc/2015/orgs/python), and 1 project from Ruby Science Foundation (https://www.google-melange.com/archive/gsoc/2015/orgs/sciruby).

Comment (gxyd) [inline doc/src/aboutus.rst]: Yes, that would be correct. I have updated it.

**Comment by gxyd** [comment]
Will merge in 24 hrs, unless @asmeurer @isuruf have any comments.
